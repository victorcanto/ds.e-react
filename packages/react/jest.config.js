module.exports = {
  roots: ['<rootDir>/src'],
  testRegex: '(/.*\\.test)\\.(ts|tsx)$',
  setupFilesAfterEnv: ['<rootDir>/jest.setup.js'],
  testEnvironment: 'jsdom',
};
