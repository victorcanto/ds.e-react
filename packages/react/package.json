{
  "name": "@ds.e/react",
  "version": "1.0.0",
  "main": "lib/index.js",
  "license": "MIT",
  "devDependencies": {
    "@babel/core": "^7.18.13",
    "@babel/preset-env": "^7.18.10",
    "@babel/preset-react": "^7.18.6",
    "@babel/preset-typescript": "^7.18.6",
    "@storybook/addon-a11y": "^6.5.10",
    "@storybook/addon-actions": "^6.5.10",
    "@storybook/addon-essentials": "^6.5.10",
    "@storybook/addon-interactions": "^6.5.10",
    "@storybook/addon-knobs": "^6.4.0",
    "@storybook/addon-links": "^6.5.10",
    "@storybook/addon-storysource": "^6.5.10",
    "@storybook/builder-webpack4": "^6.5.10",
    "@storybook/manager-webpack4": "^6.5.10",
    "@storybook/react": "^6.5.10",
    "@storybook/testing-library": "^0.0.13",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^13.3.0",
    "@types/jest": "^29.0.0",
    "@types/react": "^18.0.17",
    "babel-loader": "^8.2.5",
    "jest": "^29.0.1",
    "jest-environment-jsdom": "^29.0.1",
    "react": "^18.2.0",
    "rollup": "^2.78.1",
    "rollup-plugin-typescript2": "^0.33.0",
    "ts-node": "^10.9.1",
    "typescript": "^4.8.2"
  },
  "scripts": {
    "build": "rollup -c",
    "dev": "yarn build --watch",
    "test": "jest --verbose",
    "test:watch": "yarn test -watch",
    "storybook": "start-storybook -p 6006",
    "build-storybook": "build-storybook"
  },
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {}
}
