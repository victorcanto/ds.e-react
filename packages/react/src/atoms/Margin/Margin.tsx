import React from 'react';
import { Spacing } from '@ds.e/foundation';

export interface MarginProps {
  space?: keyof typeof Spacing;
  left?: boolean;
  right?: boolean;
  top?: boolean;
  bottom?: boolean;
  children: React.ReactNode;
}

const Margin = ({
  space = 'xxxs',
  children,
  left,
  right,
  top,
  bottom,
}: MarginProps) => {
  let className = ``;

  switch (true) {
    case left:
      className = `${className} dse-margin-left-${space}`;
      break;
    case right:
      className = `${className} dse-margin-right-${space}`;
      break;
    case top:
      className = `${className} dse-margin-top-${space}`;
      break;
    case bottom:
      className = `${className} dse-margin-bottom-${space}`;
      break;
    default:
      className = `dse-margin-${space}`;
      break;
  }

  return <div className={className}>{children}</div>;
};

export default Margin;
