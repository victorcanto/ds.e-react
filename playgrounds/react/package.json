{
  "name": "@playgrounds/react",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "devDependencies": {
    "@types/react": "^18.0.17",
    "@types/react-dom": "^18.0.6",
    "parcel": "^2.7.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "typescript": "^4.8.2"
  },
  "scripts": {
    "dev": "parcel src/index.html -p 3000"
  },
  "private": true
}
