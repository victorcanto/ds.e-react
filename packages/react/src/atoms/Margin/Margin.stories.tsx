import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { select } from '@storybook/addon-knobs';
import Margin, { MarginProps } from './Margin';

import { Spacing } from '@ds.e/foundation';

export default {
  title: 'Atoms|Margin',
} as ComponentMeta<typeof Margin>;

const Template: ComponentStory<typeof Margin> = ({ children, ...args }) => (
  <Margin {...args}>{children}</Margin>
);

export const Common = Template.bind({});
Common.args = {
  children: <p>this is some text</p>,
} as MarginProps;

export const CustomDimensions = Template.bind({});
CustomDimensions.args = {
  children: <p>this is some text</p>,
  space: select('Space', Object.values(Spacing), 'xl'),
} as MarginProps;

export const CustomPositions = Template.bind({});
CustomPositions.args = {
  children: <span>this is some text</span>,
  space: select('Space', Object.values(Spacing), 'xxl'),
  left: select('Left', [true, false], false),
  right: select('Right', [true, false], false),
  top: select('Top', [true, false], false),
  bottom: select('Bottom', [true, false], false),
} as MarginProps;
