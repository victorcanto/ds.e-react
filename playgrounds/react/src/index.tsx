import React from 'react';
import { createRoot } from 'react-dom/client';

import { Color } from '@ds.e/react';
import '@ds.e/scss/lib/Utilities.css';

const container = document.getElementById('root');
const root = createRoot(container!);
root.render(<Color hexCode='blue' width='lg' height='lg' />);
