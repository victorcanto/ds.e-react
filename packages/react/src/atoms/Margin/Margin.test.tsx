import React from 'react';
import { render } from '@testing-library/react';
import Margin from './Margin';

test('snapshot Margin atom', () => {
  const { asFragment } = render(
    <Margin left space='xl'>
      <span>this is some text</span>
    </Margin>
  );
  expect(asFragment()).toMatchSnapshot();
});
