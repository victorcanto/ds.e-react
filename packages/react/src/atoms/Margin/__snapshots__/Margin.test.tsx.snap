// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Margin atom 1`] = `
<DocumentFragment>
  <div
    class=" dse-margin-left-xl"
  >
    <span>
      this is some text
    </span>
  </div>
</DocumentFragment>
`;
